api/src/main/java/org/apache/iceberg/expressions/Expressions.java

```diff
@@ -29,17 +30,9 @@
   private Expressions() {
   }
 
-  public static Expression and(Expression left, Expression right) {
-    Preconditions.checkNotNull(left, "Left expression cannot be null.");
-    Preconditions.checkNotNull(right, "Right expression cannot be null.");
-    if (left == alwaysFalse() || right == alwaysFalse()) {
-      return alwaysFalse();
-    } else if (left == alwaysTrue()) {
-      return right;
-    } else if (right == alwaysTrue()) {
-      return left;
-    }
-    return new And(left, right);
+  public static Expression and(Expression left, Expression right, Expression... expressions) {
```

Comment:
I don't think that this is a binary compatible change. Instead, could you leave `and(left, right)` and add a new method, `and(Expression, Expression...)`? That can use `and` instead of `createAnd`.

Reply:
Good point, I'll make the change.